synapse/replication/tcp/handler.py

```diff
@@ -611,10 +611,14 @@ async def _process_position(
         # Find where we previously streamed up to.
         current_token = stream.current_token(cmd.instance_name)
 
-        # If the position token matches our current token then we're up to
-        # date and there's nothing to do. Otherwise, fetch all updates
-        # between then and now.
-        missing_updates = cmd.prev_token != current_token
+        # If the position token matches our current token then we're up to date
+        # and there's nothing to do. Otherwise, fetch all updates between then
+        # and now.
```

Comment:
```suggestion
        # If the incoming position is at least our current position then we're 
        # up to date and there's nothing to do. Otherwise, fetch all updates
        # between then and now.
```

Reply:
Example: I think the stream is at position 10.
I receive `POSITION 4 8`.
Uh oh, I might have missed stream entries 5, 6, 7 and 8. Need to rescan for them.

Reply:
Example: I think the stream is at position 10.
I receive `POSITION 12 15`.
Uh oh, I might have missed stream entries 11, 12, 13, 14, and 15. Need to rescan for them.

Reply:
Ah, no. `POSITION 4 8` means that the new token for that instance should be set to 8, and nothing has happened between 4 and 8 from that instance. (The reason for this is that normally we'd just get a stream of  `RDATA`  from the instance, which implicitly bumps the token. If for some reason the token has advanced *without* any updates to send, it sends a `POSITION`, hence why it gives the range where things have *not* happened).

Reply:
```suggestion
        # If the incoming previous position is less than our current position
	    # then we're  up to date and there's nothing to do. Otherwise, fetch
	    # all updates between then and now.
```

Maybe?